string direction
int16 speed
int16 duration
